server/asset/btc/btc.go

```diff
@@ -0,0 +1,518 @@
+// information and subscribing to block updates. It maintains a cache of block
+// data for quick lookups. BTCBackend implements asset.DEXAsset, so provides
+// exported methods for DEX-related blockchain info.
+type BTCBackend struct {
+	// An application context provided as part of the constructor. The BTCBackend
+	// will perform some cleanup when the context is cancelled.
+	ctx context.Context
+	// If an rpcclient.Client is used for the node, keeping a reference at client
+	// will result the (Client).Shutdown() being called on context cancellation.
+	client *rpcclient.Client
+	// node is used throughout for RPC calls, and in typical use will be the same
+	// as client. For testing, it can be set to a stub.
+	node btcNode
+	// The block cache stores just enough info about the blocks to shortcut future
+	// calls to GetBlockVerbose.
+	blockCache *blockCache
+	// The backend provides block notification channels through it BlockChannel
+	// method. signalMtx locks the blockChans array.
+	signalMtx   sync.RWMutex
+	blockChans  []chan uint32
+	chainParams *chaincfg.Params
+	// A logger will be provided by the dex for this backend. All logging should
+	// use the provided logger.
+	log asset.Logger
+}
+
+// Check that BTCBackend satisfies the DEXAsset interface.
+var _ asset.DEXAsset = (*BTCBackend)(nil)
+
+// NewBackend is the exported constructor by which the DEX will import the
+// backend. The provided context.Context should be cancelled when the DEX
+// application exits. The configPath can be an empty string, in which case the
+// standard system location of the bitcoind config file is assumed.
+func NewBackend(ctx context.Context, configPath string, logger asset.Logger, network asset.Network) (asset.DEXAsset, error) {
+	var params *chaincfg.Params
+	switch network {
+	case asset.Mainnet:
+		params = &chaincfg.MainNetParams
+	case asset.Testnet:
+		params = &chaincfg.TestNet3Params
+	case asset.Regtest:
+		params = &chaincfg.RegressionNetParams
+	default:
+		return nil, fmt.Errorf("unknown network ID %v", network)
+	}
+
+	if configPath == "" {
+		configPath = SystemConfigPath("bitcoin")
+	}
+
+	return NewBTCClone(ctx, configPath, logger, network, params, btcPorts)
+}
+
+// NewBTCClone creates a BTC backend for a set of network parameters and default
+// network ports. A BTC clone can use this method, possibly in conjunction with
+// ReadCloneParams, to create a BTCBackend for other assets with minimal coding.
+// See ReadCloneParams and CompatibilityCheck for more info.
+func NewBTCClone(ctx context.Context, configPath string, logger asset.Logger,
+	network asset.Network, params *chaincfg.Params, ports NetPorts) (*BTCBackend, error) {
+
+	// Read the configuration parameters
+	cfg, err := LoadConfig(configPath, network, ports)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := rpcclient.New(&rpcclient.ConnConfig{
+		HTTPPostMode: true,
+		DisableTLS:   true,
```

Comment:
I don't think so. They use a "cookie" file, which is just some randomness created at startup and deleted on shutdown.

Reply:
Hmm, that's OK I suppose, although in production it will almost certainly be necessary to run the chain daemons on separate machines, which likely means secure network connections will be required... a LAN, VPN, or tunnel can solve that easily enough though.